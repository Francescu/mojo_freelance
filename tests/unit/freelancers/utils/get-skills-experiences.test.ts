import getSkillsExperiences from "../../../../src/freelancers/utils/get-skills-experiences";

const experiences = [
  {
    id: 4,
    startDate: "2016-01-01T00:00:00+01:00",
    endDate: "2018-05-01T00:00:00+01:00",
    skills: [
      {
        id: 241,
        name: "React",
      },
      {
        id: 270,
        name: "Node.js",
      },
      {
        id: 370,
        name: "Javascript",
      },
    ],
  },
  {
    id: 54,
    startDate: "2014-01-01T00:00:00+01:00",
    endDate: "2016-09-01T00:00:00+01:00",
    skills: [
      {
        id: 470,
        name: "MySQL",
      },
      {
        id: 400,
        name: "Java",
      },
      {
        id: 370,
        name: "Javascript",
      },
    ],
  },
  {
    id: 80,
    startDate: "2013-05-01T00:00:00+01:00",
    endDate: "2014-07-01T00:00:00+01:00",
    skills: [
      {
        id: 370,
        name: "Javascript",
      },
      {
        id: 400,
        name: "Java",
      },
    ],
  },
];

describe("getSkillsExperiences", () => {
  describe("error handling", () => {
    test("It should throw an error if there is no experiences in the array", () => {
      try {
        getSkillsExperiences();
      } catch (error) {
        expect(error.message).toEqual("No experiences to process.");
      }
    });
  });

  test("It should return skills experiences", () => {
    const skillsExperiences = getSkillsExperiences(experiences);

    expect(skillsExperiences).toMatchSnapshot();
  });
});
