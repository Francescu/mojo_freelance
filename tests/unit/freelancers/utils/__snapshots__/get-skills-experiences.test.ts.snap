// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getSkillsExperiences It should return skills experiences 1`] = `
Object {
  "Java": Object {
    "dates": Array [
      Object {
        "endDate": "2016-08-31T23:00:00.000Z",
        "startDate": "2013-12-31T23:00:00.000Z",
      },
      Object {
        "endDate": "2014-06-30T23:00:00.000Z",
        "startDate": "2013-04-30T23:00:00.000Z",
      },
    ],
    "id": 400,
  },
  "Javascript": Object {
    "dates": Array [
      Object {
        "endDate": "2018-04-30T23:00:00.000Z",
        "startDate": "2015-12-31T23:00:00.000Z",
      },
      Object {
        "endDate": "2016-08-31T23:00:00.000Z",
        "startDate": "2013-12-31T23:00:00.000Z",
      },
      Object {
        "endDate": "2014-06-30T23:00:00.000Z",
        "startDate": "2013-04-30T23:00:00.000Z",
      },
    ],
    "id": 370,
  },
  "MySQL": Object {
    "dates": Array [
      Object {
        "endDate": "2016-08-31T23:00:00.000Z",
        "startDate": "2013-12-31T23:00:00.000Z",
      },
    ],
    "id": 470,
  },
  "Node.js": Object {
    "dates": Array [
      Object {
        "endDate": "2018-04-30T23:00:00.000Z",
        "startDate": "2015-12-31T23:00:00.000Z",
      },
    ],
    "id": 270,
  },
  "React": Object {
    "dates": Array [
      Object {
        "endDate": "2018-04-30T23:00:00.000Z",
        "startDate": "2015-12-31T23:00:00.000Z",
      },
    ],
    "id": 241,
  },
}
`;
